scipy/linalg/_special_matrices.py

```diff
@@ -308,6 +309,10 @@ def kron(a, b):
     """
     Kronecker product.
 
+    .. deprecated:: 1.13.0
+        `kron` is deprecated and will be removed in SciPy 1.15.0. Use
+        `numpy.kron` instead.
```

Comment:
The discussion on whether we are following https://numpy.org/neps/nep-0023-backwards-compatibility.html#implementing-deprecations-and-removals and not specifying a release for removal in gh-19880 seems unresolved, so should probably consider it here.